### ISSUE TITLE
Tests with django sites framework onetoonefield causes unexpected behavior

### ISSUE THREAD

**Issue by peterlauri** [pytest-dev/pytest-django#322]
Assume you have a model:

```
class Customer(models.Model):
    site = models.OneToOneField('sites.Site')
```

And when using the sites middleware, without setting SITE_ID, the site is looked up and cached based on the requests host information: https://github.com/django/django/blob/master/django/contrib/sites/models.py#L12

This causes unexpected behavior if testing a multi tenant site, as the request.site object will be the one from the SITE_CACHE, that might have an already populated _request.site.customer_ from previous execution.

I will submit a proposal for fixing this, as it can cause plenty of pain when debugging :)

**Comment by blueyed** [comment]
Is this covered / fixed by #324 maybe?

**Comment by peterlauri** [comment]
@blueyed I looked into all receivers of **django.test.signals.setting_changed** in django project, and non looks to cover reset of SITE_CACHE. So I don't think #324 will fix this.
